types/universalify/index.d.ts

```diff
@@ -1,7 +1,21 @@
-// Type definitions for universalify 1.0
+// Type definitions for universalify 2.0
 // Project: https://github.com/RyanZim/universalify#readme
-// Definitions by: Richie Bendall <https://github.com/Richienb>
+// Definitions by: Richie Bendall <https://github.com/Richienb>, Jan Peer Stöcklmair <https://github.com/JPeer264>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.7
 
-export function fromCallback(fn: (...args: any[]) => any): (...args: any[]) => Promise<any> | void;
-export function fromPromise(fn: (...args: any[]) => any): (...args: any[]) => Promise<any> | void;
+import { List } from 'ts-toolbelt';
+
+export interface UniversalifyResult<T, A extends any[]> {
```

Comment:
We should give these types more descriptive names like `ReturnValueType` and `ArgumentsType`